51/clamytoe/nba.py

```diff
@@ -0,0 +1,178 @@
+from collections import namedtuple
+from time import time
+import csv
+import os
+import sqlite3
+
+import requests
+
+DATA_URL = 'https://query.data.world/s/ezwk64ej624qyverrw6x7od7co7ftm'
+DATA_CACHED = 'nba.data'
+NBA_DB = 'nba.db'
+
+# start clean
+if os.path.isfile(NBA_DB):
+    os.remove(NBA_DB)
+
+Player = namedtuple('Player', ('name year first_year team college active '
+                               'games avg_min avg_points'))
+
+conn = sqlite3.connect(NBA_DB)
+cur = conn.cursor()
+
+
+def _get_csv_data():
+    """GIVEN:
+       Load in CSV data in from remote URL or local cache file"""
+    if os.path.isfile(DATA_CACHED):
+        with open(DATA_CACHED) as f:
+            return f.read()
+    else:
+        with requests.Session() as session:
+            return session.get(DATA_URL).content.decode('utf-8')
+
+
+def load_data():
+    """GIVEN:
+       Converts NBA CSV data into a list of Player namedtuples"""
+    content = _get_csv_data()
+    reader = csv.DictReader(content.splitlines(), delimiter=',')
+    for row in reader:
+        player = Player(name=row['Player'],
+                        year=row['Draft_Yr'],
+                        first_year=row['first_year'],
+                        team=row['Team'],
+                        college=row['College'],
+                        active=row['Yrs'],
+                        games=row['Games'],
+                        avg_min=row['Minutes.per.Game'],
+                        avg_points=row['Points.per.Game'])
+        yield player
+
+
+# CODE HERE (tests under __main__):
+
+def import_to_db(players=None):
+    """Create database table in sqlite3 and import the players data
+
+       required table SQL:
+       CREATE TABLE players (name, year, first_year, team, college,
+                             active, games, avg_min, avg_points)
+    """
+    create_players_table = """
+    -- players table
+    CREATE TABLE IF NOT EXISTS players (
+        name text NOT NULL,
+        year int NOT NULL,
+        first_year int NOT_NULL,
+        team text NOT NULL,
+        college text NOT NULL,
+        active int NOT NULL,
+        games int NOT NULL,
+        avg_min float NOT NULL,
```

Comment:
Did you find out what made the create slow, was it the typing? I do like you added the column types here!